include/lbann/layers/transform/slice_impl.hpp

```diff
@@ -0,0 +1,134 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2023, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef LBANN_LAYER_SLICE_IMPL_HPP_INCLUDED
+#define LBANN_LAYER_SLICE_IMPL_HPP_INCLUDED
+
+#include "lbann/data_coordinator/data_coordinator.hpp"
+#include "lbann/layers/transform/slice.hpp"
+
+namespace lbann {
+
+template <typename TensorDataType, data_layout Layout, El::Device Device>
+void slice_layer<TensorDataType, Layout, Device>::setup_dims()
+{
+  data_type_layer<TensorDataType>::setup_dims();
+
+  // Setup the slice points if they are to be established by the data reader
+  // TODO: Move this responsibility to another component (input layer)
+  if (m_set_slice_points_from_data_reader) {
+    std::vector<size_t> slice_points;
+    std::string slice_point_method_name = "'get_slice_points_from_reader'";
+
+    LBANN_WARNING("slice_points_from_reader is deprecated and will be removed "
+                  "in a future version.");
```

Comment:
apart from refactoring to the impl file, these lines are the extent of the changes made.